src/controllers/auth-controller.spec.ts

```diff
@@ -430,4 +447,192 @@ describe('Auth', () => {
+        .send({ token: firstUserResetToken });
+      const token = await resetTokenRepository.findOne({ token: firstUserResetToken });
+      expect(res.status).toEqual(400);
+      expect(token).toBeDefined();
+      expect(token!.token).toEqual(firstUserResetToken);
+      expect(JSON.parse(res.text).error).toEqual('Reset token and passwords are required');
+      expect(spyOnSendResetCompleteMail).not.toHaveBeenCalled();
+    });
+
+    it('not be able update password without reset token provided', async () => {
+      const res = await request(app)
+        .post(urls.reset)
+        .set('Accept', 'application/json')
+        .send({ password: firstUserPassword });
+      const token = await resetTokenRepository.findOne({ token: firstUserResetToken });
+      expect(res.status).toEqual(400);
+      expect(token).toBeDefined();
+      expect(token!.token).toEqual(firstUserResetToken);
+      expect(JSON.parse(res.text).error).toEqual('Reset token and passwords are required');
+      expect(spyOnSendResetCompleteMail).not.toHaveBeenCalled();
+    });
+
+    it('be able to update password with passed token and new password; and by this disable used token', async () => {
+      const response = await request(app)
+        .post(urls.reset)
+        .set('Accept', 'application/json')
+        .send({ token: firstUserResetToken, password: firstUserNewPassword });
+
+      const sameToken = await resetTokenRepository.findOne({ userId: firstUser.id });
+      const sameUser = await userRepository.findOne({ id: firstUser.id });
+
+      if (!sameUser) {
+        throw new Error('user should exist');
+      }
+
+      expect(response.status).toEqual(200);
+      expect(response.body.ok).toEqual(true);
+      expect(response.status).toEqual(200);
+      expect(sameToken).not.toBeDefined();
+      expect(sameUser).toBeDefined();
+      expect(spyOnSendResetCompleteMail).toHaveBeenCalled();
+
+      const loginResponse = await request(app)
+        .post(urls.login)
+        .set('Accept', 'application/json')
+        .send({ email: firstUserEmail, password: firstUserNewPassword });
+
+      expect(loginResponse.status).toEqual(200);
+    });
+
+    it('be not able to reset password by the same reset token twice', async () => {
+      const res = await request(app)
+        .post(urls.reset)
+        .set('Accept', 'application/json')
+        .send({ token: firstUserResetToken, password: firstUserPassword });
+
+      const token = await resetTokenRepository.findOne({ token: firstUserResetToken });
+
+      expect(res.status).toEqual(401);
+      expect(token).not.toBeDefined();
+      expect(JSON.parse(res.text).error).toEqual('Token already was used or never existed');
+      expect(spyOnSendResetCompleteMail).not.toHaveBeenCalled();
+
+      const loginResponse = await request(app)
+        .post(urls.login)
+        .set('Accept', 'application/json')
+        .send({ email: firstUserEmail, password: firstUserNewPassword });
```

Comment:
I find this check ridiculous -- previous successful suite already confirms that login is achievable